packages/lib/test/treeUtils/sandbox.test.ts

```diff
@@ -342,3 +345,103 @@ test("sandbox cannot be changed outside of fn", () => {
 
   expect(() => n.setValue(1)).toThrow("tried to invoke action 'setValue' over a readonly node")
 })
+
+test("sanboxed nodes can check if they are sandboxed", () => {
+  const initEvents: string[] = []
+
+  @model("A2")
+  class A2 extends Model({
+    b: prop<B2>(),
+  }) {
+    onInit() {
+      initEvents.push(`A2 init: ${isSandboxedNode(this)}`)
+    }
+
+    onAttachedToRootStore() {
+      initEvents.push(`A2 attached: ${isSandboxedNode(this)}`)
+    }
+
+    @computed
+    get isSandboxed() {
+      return isSandboxedNode(this)
+    }
+
+    @modelAction
+    shouldBeSandboxed(shouldBeSandboxed: boolean): void {
+      expect(isSandboxedNode(this)).toBe(shouldBeSandboxed)
+    }
+  }
+
+  @model("B2")
+  class B2 extends Model({
+    value: prop<number>(),
+  }) {
+    onInit() {
+      initEvents.push(`B2 init: ${isSandboxedNode(this)}`)
+    }
+
+    onAttachedToRootStore() {
+      initEvents.push(`B2 attached: ${isSandboxedNode(this)}`)
+    }
+
+    @computed
+    get isSandboxed() {
+      return isSandboxedNode(this)
+    }
+
+    @modelAction
+    shouldBeSandboxed(shouldBeSandboxed: boolean): void {
+      expect(isSandboxedNode(this)).toBe(shouldBeSandboxed)
+    }
+  }
+
+  const a = new A2({ b: new B2({ value: 0 }) })
```

Comment:
Should `a` be unregistered as a root store after the test has completed?

```ts
autoDispose(() => {
  if (isRootStore(a)) {
    unregisterRootStore(a)
  }
})
```

Reply:
done